ttnn/cpp/ttnn/operations/pool/grid_sample/device/grid_sample_op.cpp

```diff
@@ -86,10 +92,14 @@ void GridSample::validate(const std::vector<Tensor>& input_tensors) const {
     TT_FATAL(input_tensor.layout() == Layout::ROW_MAJOR, "Input tensor must be ROW_MAJOR layout");
     TT_FATAL(grid_tensor.layout() == Layout::ROW_MAJOR, "Grid tensor must be ROW_MAJOR layout");
 
-    // Parameter validation - currently only support fixed configuration
-    TT_FATAL(mode_ == "bilinear", "Only bilinear interpolation mode is currently supported");
     TT_FATAL(padding_mode_ == "zeros", "Only zeros padding mode is currently supported");
 
+    // Mode and precomputed grid compatibility validation
+    TT_FATAL(
```

Comment:
do we need additonal checks for nearest to validate which tensor types are supported

Reply:
if nearest supports just precomputd grid let's remove code for path where grid is not precomputed for now